sdk/src/client.ts

```diff
@@ -8,21 +8,24 @@ export interface DirectusClient<Schema extends object> {
 	use: <Extension extends object>(createExtension: (client: DirectusClient<Schema>) => Extension) => this & Extension;
 }
 
-export const useDirectus = <Schema extends object = any>(url: string | URL, config?: ClientConfig) => {
-	const client: DirectusClient<Schema> = {
-		url: typeof url === 'string' ? new URL(url) : url,
-		auth: {
-			token: config?.token ?? null,
-		},
-		use(createExtension) {
-			const extension = createExtension(this);
-
-			return {
-				...this,
-				...extension,
-			};
-		},
-	};
-
-	return client;
-};
+/**
+ * Creates a client to communicate with a Directus app.
+ *
+ * @param url The URL to the Directus app.
+ * @param config The optional configuration.
+ *
+ * @returns A Directus client.
+ */
+export const useDirectus = <Schema extends object = any>(
+	url: string,
+	config?: ClientConfig | undefined
+): DirectusClient<Schema> => ({
+	url: new URL(url),
+	auth: { token: config?.token ?? null },
+	use(createExtension) {
+		return {
+			...this,
+			...createExtension(this),
+		};
+	},
+});
```

Comment:
A lot of this refactoring is at the cost of readability and in my opinion and feels like pre-mature optimization at this stage to save a total of 20 Bytes in the resulting bundle (or 8bytes  compared to this suggestion) 😬 
```suggestion
export const useDirectus = <Schema extends object = any>(
	url: string,
	config?: ClientConfig | undefined
): DirectusClient<Schema> => {
	return {
		url: new URL(url),
		auth: {
			token: config?.token ?? null,
		},
		use(createExtension) {
			const extension = createExtension(this);

			return {
				...this,
				...extension,
			};
		},
	};
};
```